src/Raven.Server/Documents/Queries/AST/QueryExpression.cs

```diff
@@ -32,7 +32,7 @@ private static object Evaluate(QueryExpression q, BlittableJsonReaderObject valu
             }
         }
 
-        public static bool IsMatchedBy(this QueryExpression where, BlittableJsonReaderObject value, BlittableJsonReaderObject queryParameters)
+        public static bool IsMatchedBy(this QueryExpression where, BlittableJsonReaderObject value, BlittableJsonReaderObject queryParameters, bool shouldCaseSensitiveStringCompare = false)
```

Comment:
Future note, this should be changed to an options object if/when we'll have more state here